internal/pkg/errors/error_context.go

```diff
@@ -13,8 +13,20 @@ type ErrorContext struct {
 // NewErrorContext creates an empty ErrorContext.
 func NewErrorContext() *ErrorContext { return &ErrorContext{} }
 
-// Add formats and appends the passed prefix and value onto the error contexts.
+// Add formats and upserts the passed prefix and value onto the error contexts.
 func (ctx *ErrorContext) Add(prefix, val string) {
+	idx := -1
+	for i, c := range ctx.contexts {
+		if strings.HasPrefix(c, prefix) {
+			idx = i
+			break
```

Comment:
Isn't it possible to `ctx.contexts[idx] = prefix + val` here directly and then `break`?

(and same for `ui_context.go`)

Reply:
It is possible, however, in that case, the error context value is malformed.  The current concatenated value that is stored into context makes this issue very difficult to tease apart.  A future goal is to split these context values into a key/value pair, which would give us an opportunity to make this less _brittle_.